File: limeutils/src/main/java/com/xlib/limeutils/network/NetworkUtils.java
package com.xlib.limeutils.network;

import android.annotation.SuppressLint;
import android.content.Context;
import android.database.Cursor;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.provider.ContactsContract;
import android.telephony.TelephonyManager;

import com.xlib.limeutils.base.Contextor;


/**
 * Created by Xplo on 1/4/2016.
 */
public class NetworkUtils {

    private static final String TAG = "NetworkUtils";

    /**
     * Method to check internet is avalable
     * need permission in manifest
     *
     * @param context
     * @return
     */
    public static boolean isConnectedToInternet(Context context) {
        ConnectivityManager connectivity = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connectivity != null) {
            @SuppressLint("MissingPermission") NetworkInfo[] info = connectivity.getAllNetworkInfo();
            if (info != null)
                for (int i = 0; i < info.length; i++)
                    if (info[i].getState() == NetworkInfo.State.CONNECTED) {
                        return true;
                    }

        }
        return false;
    }


    public static String getSimNumber(Context context) {
        TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
        //String getSimSerialNumber = telephonyManager.getSimSerialNumber();
        @SuppressLint("MissingPermission") String getSimNumber = telephonyManager.getLine1Number();

        return getSimNumber;
    }

    public static String getSimSerial(Context context) {
        TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
        @SuppressLint("MissingPermission") String getSimSerialNumber = telephonyManager.getSimSerialNumber();
        @SuppressLint("MissingPermission") String getSimNumber = telephonyManager.getLine1Number();


        return getSimSerialNumber;
    }

    public static String getIMEI(Context context) {
        TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
        @SuppressLint("MissingPermission") String IMEI = telephonyManager.getDeviceId();

        return IMEI;
    }

}

File: limeutils/src/main/java/com/xlib/limeutils/network/NetworkUtils.kt
package com.xlib.limeutils.network

import android.annotation.SuppressLint
import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.telephony.TelephonyManager


/**
 * Created by Xplo on 1/4/2016.
 */
@SuppressLint("MissingPermission")
object NetworkUtils {

    private const val TAG = "NetworkUtils"

    /**
     * Method to check internet is avalable
     * need permission in manifest
     *
     * @param context
     * @return
     */
    fun isNetworkConnected(context: Context): Boolean {
        val connectivity = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        if (connectivity != null) {
            val info = connectivity.allNetworkInfo
            if (info != null)
                for (i in info.indices)
                    if (info[i].state == NetworkInfo.State.CONNECTED) {
                        return true
                    }

        }
        return false
    }


    fun getSimNumber(context: Context): String {
        val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
        //String getSimSerialNumber = telephonyManager.getSimSerialNumber();

        return telephonyManager.line1Number
    }

    fun getSimSerial(context: Context): String {
        val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
        val getSimSerialNumber = telephonyManager.simSerialNumber
        val getSimNumber = telephonyManager.line1Number


        return getSimSerialNumber
    }

    fun getIMEI(context: Context): String {
        val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager

        return telephonyManager.deviceId
    }

}
